nautobot/extras/utils.py

```diff
@@ -648,3 +650,62 @@ def bulk_delete_with_bulk_change_logging(qs, batch_size=1000):
         finally:
             change_context.defer_object_changes = False
             change_context.reset_deferred_object_changes()
+
+
+def create_schedule(schedule_data, job_kwargs, job_model, user, approval_required, task_queue=None):
+    """
+    This is an internal function to create a scheduled job from API data.
+    It has to handle both once-offs (i.e. of type TYPE_FUTURE) and interval
+    jobs.
+    """
+    from nautobot.extras.models.jobs import ScheduledJob  # avoid circular import
```

Comment:
Would it make more sense to implement this as a method on the `ScheduledJob` model class, similar to how we have `JobResult.enqueue_job()`, have `ScheduledJob.create_schedule()`?